runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineResult.java

```diff
@@ -207,12 +202,19 @@ protected State awaitTermination(final Duration duration) throws ExecutionExcept
            terminationState = State.DONE;
            break;
          default:
-           state = State.UNKNOWN;
+           terminationState = State.UNKNOWN;
            break;
        }
        return terminationState;
     }
+  }
 
+  private void changeState(State newState) {
```

Comment:
But all calls end up finalizing the state from what I saw. Again, totally up to you.